src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -3646,4 +3646,175 @@ public Long hstrlen(final String key, final String field) {
     return client.getIntegerReply();
   }
 
+  @Override
+  public StreamnEntryID xadd(final String key, final StreamnEntryID id, final Map<String, String> hash) {
+    return xadd(key, id, hash, Long.MAX_VALUE, true);
+  }
+  
+  @Override
+  public StreamnEntryID xadd(final String key, StreamnEntryID id, final Map<String, String> hash, final long maxLen, final boolean exactMaxLen) {
```

Comment:
IMO, default behavior should be how the command behaves without that parameter.

In Redis, extra `~` means approximate length. `approximateLength` is better parameter name than `exactMaxLen`. This way it can expect `false` by default. Moreover, isn't `false` default for:
1. xgroupCreate(boolean makeStream)
2. xreadGroup(boolean noAck)
3. xclaim(boolean force)

Reply:
notice I didn't create an overload without the "exactMaxLen", these are the two options I gave, so when a user is picking maxLen he has to consider the exactMaxLen and not assume anything about it.
xadd(key, id, hash)
xadd(key, id, hash, maxLen, exactMaxLen)

Reply:
done